torch/csrc/jit/passes/symbolic_shape_cache.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include <torch/csrc/jit/ir/ir.h>
+#include <torch/csrc/jit/passes/symbolic_shape_analysis.h>
+
+namespace torch {
+namespace jit {
+
+struct TORCH_API CanonicalizedSymbolicShape {
+  CanonicalizedSymbolicShape(
+      const c10::SymbolicShape& orig_shape,
+      std::unordered_map<int64_t, int64_t>& ss_map) {
+    init(orig_shape, ss_map);
+  }
+
+  CanonicalizedSymbolicShape(c10::SymbolicShape& orig_shape) {
```

Comment:
When would this ever get called without `ss_map` testing ?

Reply:
Yeah, currently it is used in testing. I figure it is valid to want to canonicalize without having some reference map, as it was useful for testing purposes.